src/dvclive/live.py

```diff
@@ -148,28 +150,36 @@ def _init_cleanup(self):
             os.remove(dvc_file)
 
     @catch_and_warn(DvcException, logger)
-    def _init_dvc(self):
+    def _init_dvc(self):  # noqa: C901
         from dvc.scm import NoSCM
 
         if os.getenv(env.DVC_ROOT, None):
             self._inside_dvc_pipeline = True
             self._init_dvc_pipeline()
         self._dvc_repo = get_dvc_repo()
 
+        scm = self._dvc_repo.scm if self._dvc_repo else None
+        if isinstance(scm, NoSCM):
+            scm = None
+        if scm:
+            self._baseline_rev = scm.get_rev()
+        self._exp_name = get_exp_name(self._exp_name, scm, self._baseline_rev)
+        logger.info(f"Logging to experiment '{self._exp_name}'")
```

Comment:
Once we have tried to find a repo, we can generate all exp info we need instead of scattering it throughout the code